yarn-project/p2p/src/mem_pools/tx_pool/aztec_kv_tx_pool.ts

```diff
@@ -569,183 +638,47 @@ export class AztecKVTxPool extends (EventEmitter as new () => TypedEventEmitter<
-   *   - txs with an expiration timestamp lower than that of the mined block
-   *
-   * @param minedTxHashes - The tx hashes of the txs mined in the block.
-   * @param blockHeader - The header of the mined block.
-   * @returns The total number of txs evicted from the pool.
-   */
-  private async evictInvalidTxsAfterMining(
-    minedTxHashes: TxHash[],
-    blockHeader: BlockHeader,
-    minedNullifiers: Set<string>,
-    minedFeePayers: Set<string>,
-  ): Promise<number> {
-    if (minedTxHashes.length === 0) {
-      return 0;
-    }
-
-    const { blockNumber, timestamp } = blockHeader.globalVariables;
-
-    // Wait for world state to be synced to at least the mined block number
-    await this.#worldStateSynchronizer.syncImmediate(blockNumber);
-
-    const db = this.#worldStateSynchronizer.getCommitted();
-    const gasTxValidator = this.createGasTxValidator(db);
-
-    const txsToEvict: TxHash[] = [];
-    for await (const txHash of this.#pendingTxPriorityToHash.valuesAsync()) {
-      const tx = await this.getPendingTxByHash(txHash);
-      if (!tx) {
-        continue;
-      }
-
-      // Evict pending txs that share nullifiers with mined txs
-      const txNullifiers = tx.data.getNonEmptyNullifiers();
-      if (txNullifiers.some(nullifier => minedNullifiers.has(nullifier.toString()))) {
-        this.#log.verbose(`Evicting tx ${txHash} from pool due to a duplicate nullifier with a mined tx`);
-        txsToEvict.push(TxHash.fromString(txHash));
-        continue;
-      }
-
-      // Evict pending txs with an insufficient fee payer balance
-      if (
-        minedFeePayers.has(tx.data.feePayer.toString()) &&
-        (await gasTxValidator.validateTxFee(tx)).result === 'invalid'
-      ) {
-        this.#log.verbose(`Evicting tx ${txHash} from pool due to an insufficient fee payer balance`);
-        txsToEvict.push(TxHash.fromString(txHash));
-        continue;
-      }
-
-      // Evict pending txs with an expiration timestamp less than or equal to the mined block timestamp
-      const includeByTimestamp = tx.data.includeByTimestamp;
-      if (includeByTimestamp <= timestamp) {
-        this.#log.verbose(
-          `Evicting tx ${txHash} from pool due to the tx being expired (includeByTimestamp: ${includeByTimestamp}, mined block timestamp: ${timestamp})`,
-        );
-        txsToEvict.push(TxHash.fromString(txHash));
-        continue;
-      }
-    }
-
-    if (txsToEvict.length > 0) {
-      await this.deleteTxs(txsToEvict, { eviction: true });
-    }
-    return txsToEvict.length;
+  private async removePendingTxIndices(tx: Tx, txHash: string): Promise<void> {
+    return await this.#store.transactionAsync(async () => {
+      await this.#pendingTxPriorityToHash.deleteValue(getPendingTxPriority(tx), txHash);
+      this.#pendingTxs.delete(txHash);
+      await this.#historicalHeaderToTxHash.deleteValue(
+        (await tx.data.constants.anchorBlockHeader.hash()).toString(),
```

Comment:
I wonder if computing the anchor block header hash so many times could have a performance impact. Let's not optimize prematurely, but we should keep an eye on it if we see the tx pool is slow-ish.